pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogic_role failed

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_role [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15563158?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15563158?buildTab=artifacts#/) on master @ [bd0659d192bf69a952733e386ddb205ea9ae491d](https://github.com/cockroachdb/cockroach/commits/bd0659d192bf69a952733e386ddb205ea9ae491d):


```
=== RUN   TestTenantLogic_role
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/40b2f96e9dfb8d2f80118e4c45bc2e13/logTestTenantLogic_role3302268835
    test_log_scope.go:81: use -show-logs to present logs inline
[21:00:37] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/14698/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/role: 43 statements
[21:00:43] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/14698/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/role: 44 statements
[21:00:45] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/14698/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/role: 92 statements
    logic.go:2968: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/14698/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/role:465: SELECT * FROM system.role_members
        expected:
            role   member    isAdmin  role_id  member_id
            admin  root      true     2        1
            rolea  roleb     true     111      112
            rolea  rolee     false    111      115
            roleb  rolec     false    112      113
            rolec  roled     false    113      114
            rolec  testuser  false    113      110
            roled  testuser  false    114      110
            
        but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
            role   member    isAdmin  role_id  member_id
            admin  root      true     2        1
            rolea  roleb     true     112      113
            rolea  rolee     false    112      116
            roleb  rolec     false    113      114
            rolec  roled     false    114      115
            rolec  testuser  false    114      111
            roled  testuser  false    115      111
            
[21:00:45] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/14698/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/role with config 3node-tenant: 96 tests, 1 failures
    logic.go:4135: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/14698/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/role:477: error while processing
    logic.go:4135: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/14698/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/role:477: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/40b2f96e9dfb8d2f80118e4c45bc2e13/logTestTenantLogic_role3302268835
--- FAIL: TestTenantLogic_role (12.61s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_role.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39399

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogic_role [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15592196?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15592196?buildTab=artifacts#/) on master @ [591eec9f33a36d0749ec87c6c3141dd275eae81f](https://github.com/cockroachdb/cockroach/commits/591eec9f33a36d0749ec87c6c3141dd275eae81f):


```
=== RUN   TestTenantLogic_role
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/40b2f96e9dfb8d2f80118e4c45bc2e13/logTestTenantLogic_role4021317645
    test_log_scope.go:81: use -show-logs to present logs inline
[16:28:14] setting distsql_workmem='77242B';
[16:28:14] setting distsql_workmem='77242B';
[16:28:23] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3254/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/role: 44 statements
[16:28:25] --- progress: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3254/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/role: 89 statements
    logic.go:2968: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3254/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/role:465: SELECT * FROM system.role_members
        expected:
            role   member    isAdmin  role_id  member_id
            admin  root      true     2        1
            rolea  roleb     true     111      112
            rolea  rolee     false    111      115
            roleb  rolec     false    112      113
            rolec  roled     false    113      114
            rolec  testuser  false    113      110
            roled  testuser  false    114      110
            
        but found (query options: "colnames,rowsort" -> ignore the following ordering of rows) :
            role   member    isAdmin  role_id  member_id
            admin  root      true     2        1
            rolea  roleb     true     112      113
            rolea  rolee     false    112      116
            roleb  rolec     false    113      114
            rolec  roled     false    114      115
            rolec  testuser  false    114      111
            roled  testuser  false    115      111
            
[16:28:25] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3254/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/role with config 3node-tenant: 96 tests, 1 failures
    logic.go:4135: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3254/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/role:477: error while processing
    logic.go:4135: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3254/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/role:477: too many errors encountered, skipping the rest of the input
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/40b2f96e9dfb8d2f80118e4c45bc2e13/logTestTenantLogic_role4021317645
--- FAIL: TestTenantLogic_role (12.70s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogic_role.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/sql/logictest/testdata/logic_test/role
+++ b/pkg/sql/logictest/testdata/logic_test/role
@@ -157,12 +157,12 @@ user root
 statement ok
 GRANT testrole TO testuser
 
-query TTBOO colnames,rowsort
-SELECT * FROM system.role_members
+query TTB colnames,rowsort
+SELECT role, member, "isAdmin" FROM system.role_members
 ----
-role      member    isAdmin  role_id  member_id
-admin     root      true     2        1
-testrole  testuser  false    108      100
+role      member    isAdmin
+admin     root      true
+testrole  testuser  false
 
 query TTB colnames,rowsort
 SHOW GRANTS ON ROLE
@@ -186,14 +186,14 @@ CREATE ROLE child_role;
 GRANT testrole to child_role;
 GRANT testuser TO child_role;
 
-query TTBOO colnames,rowsort
-SELECT * FROM system.role_members
+query TTB colnames,rowsort
+SELECT role, member, "isAdmin" FROM system.role_members
 ----
-role      member      isAdmin  role_id  member_id
-admin     root        true     2        1
-testrole  child_role  false    108      109
-testrole  testuser    true     108      100
-testuser  child_role  false    100      109
+role      member      isAdmin
+admin     root        true
+testrole  child_role  false
+testrole  testuser    true
+testuser  child_role  false
 
 query TTBB colnames,rowsort
 SELECT * FROM "".crdb_internal.kv_inherited_role_members
@@ -258,14 +258,14 @@ statement ok
 GRANT admin TO testuser
 
 # Dropping users/roles deletes all their memberships.
-query TTBOO colnames,rowsort
-SELECT * FROM system.role_members
+query TTB colnames,rowsort
+SELECT role, member, "isAdmin" FROM system.role_members
 ----
-role      member     isAdmin  role_id  member_id
-admin     root       true     2        1
-admin     testuser   false    2        100
-testrole  testuser   true     108      100
-testrole  testuser2  true     108      107
+role      member     isAdmin
+admin     root       true
+admin     testuser   false
+testrole  testuser   true
+testrole  testuser2  true
 
 query TTB colnames,rowsort
 SHOW GRANTS ON ROLE
@@ -324,21 +324,21 @@ DROP USER testuser
 statement ok
 CREATE USER testuser
 
-query TTBOO colnames,rowsort
-SELECT * FROM system.role_members
+query TTB colnames,rowsort
+SELECT role, member, "isAdmin" FROM system.role_members
 ----
-role      member     isAdmin  role_id  member_id
-admin     root       true     2        1
-testrole  testuser2  true     108      107
+role      member     isAdmin
+admin     root       true
+testrole  testuser2  true
 
 statement ok
 DROP ROLE testrole
 
-query TTBOO colnames
-SELECT * FROM system.role_members
+query TTB colnames
+SELECT role, member, "isAdmin" FROM system.role_members
 ----
-role   member  isAdmin  role_id  member_id
-admin  root    true     2        1
+role   member  isAdmin
+admin  root    true
 
 # Test cycle detection.
 statement error pq: admin cannot be a member of itself
@@ -462,17 +462,17 @@ testuser
 
 user root
 
-query TTBOO colnames,rowsort
-SELECT * FROM system.role_members
+query TTB colnames,rowsort
+SELECT role, member, "isAdmin" FROM system.role_members
 ----
-role   member    isAdmin  role_id  member_id
-admin  root      true     2        1
-rolea  roleb     true     111      112
-rolea  rolee     false    111      115
-roleb  rolec     false    112      113
-rolec  roled     false    113      114
-rolec  testuser  false    113      110
-roled  testuser  false    114      110
+role   member    isAdmin
+admin  root      true
+rolea  roleb     true
+rolea  rolee     false
+roleb  rolec     false
+rolec  roled     false
+rolec  testuser  false
+roled  testuser  false
 
 statement ok
 DROP ROLE rolea
@@ -480,12 +480,12 @@ DROP ROLE rolea
 statement ok
 DROP ROLE rolec
 
-query TTBOO colnames,rowsort
-SELECT * FROM system.role_members
+query TTB colnames,rowsort
+SELECT role, member, "isAdmin" FROM system.role_members
 ----
-role   member    isAdmin  role_id  member_id
-admin  root      true     2        1
-roled  testuser  false    114      110
+role   member    isAdmin
+admin  root      true
+roled  testuser  false
 
 query TTT rowsort
 SHOW ROLES
@@ -546,13 +546,13 @@ GRANT "" TO rolea
 statement error role/user "" does not exist
 REVOKE "" FROM rolea
 
-query TTBOO colnames,rowsort
-SELECT * FROM system.role_members
+query TTB colnames,rowsort
+SELECT role, member, "isAdmin" FROM system.role_members
 ----
-role   member    isAdmin  role_id  member_id
-admin  root      true     2        1
-rolea  testuser  true     116      110
-roleb  testuser  true     117      110
+role   member    isAdmin
+admin  root      true
+rolea  testuser  true
+roleb  testuser  true
 
 user testuser
 
@@ -561,15 +561,15 @@ GRANT rolea,roleb TO root WITH ADMIN OPTION
 
 user root
 
-query TTBOO colnames,rowsort
-SELECT * FROM system.role_members
+query TTB colnames,rowsort
+SELECT role, member, "isAdmin" FROM system.role_members
 ----
-role   member    isAdmin  role_id  member_id
-admin  root      true     2        1
-rolea  root      true     116      1
-rolea  testuser  true     116      110
-roleb  root      true     117      1
-roleb  testuser  true     117      110
+role   member    isAdmin
+admin  root      true
+rolea  root      true
+rolea  testuser  true
+roleb  root      true
+roleb  testuser  true
 
 query TTT colnames,rowsort
 SELECT * FROM information_schema.administrable_role_authorizations
@@ -631,45 +631,45 @@ REVOKE roleb FROM root
 
 user root
 
-query TTBOO colnames,rowsort
-SELECT * FROM system.role_members
+query TTB colnames,rowsort
+SELECT role, member, "isAdmin" FROM system.role_members
 ----
-role   member    isAdmin  role_id  member_id
-admin  root      true     2        1
-rolea  root      true     116      1
-rolea  testuser  false    116      110
-roleb  testuser  true     117      110
+role   member    isAdmin
+admin  root      true
+rolea  root      true
+rolea  testuser  false
+roleb  testuser  true
 
 statement ok
 REVOKE rolea, roleb FROM testuser, root
 
-query TTBOO colnames
-SELECT * FROM system.role_members
+query TTB colnames
+SELECT role, member, "isAdmin" FROM system.role_members
 ----
-role   member  isAdmin  role_id  member_id
-admin  root    true     2        1
+role   member  isAdmin
+admin  root    true
 
 # Verify that GRANT/REVOKE are not sensitive to the case of role names.
 
 statement ok
 GRANT roLea,rOleB TO tEstUSER WITH ADMIN OPTION
 
-query TTBOO colnames,rowsort
-SELECT * FROM system.role_members
+query TTB colnames,rowsort
+SELECT role, member, "isAdmin" FROM system.role_members
 ----
-role   member    isAdmin  role_id  member_id
-admin  root      true     2        1
-rolea  testuser  true     116      110
-roleb  testuser  true     117      110
+role   member    isAdmin
+admin  root      true
+rolea  testuser  true
+roleb  testuser  true
 
 statement ok
 REVOKE roleA, roleB FROM TestUser
 
-query TTBOO colnames
-SELECT * FROM system.role_members
+query TTB colnames
+SELECT role, member, "isAdmin" FROM system.role_members
 ----
-role   member  isAdmin  role_id  member_id
-admin  root    true     2        1
+role   member  isAdmin
+admin  root    true
 
 # Test privilege checks.
 
@@ -700,13 +700,13 @@ GRANT admin TO newgroup
 
 user testuser
 
-query TTBOO colnames,rowsort
-SELECT * FROM system.role_members
+query TTB colnames,rowsort
+SELECT role, member, "isAdmin" FROM system.role_members
 ----
-role      member    isAdmin  role_id  member_id
-admin     newgroup  false    2        118
-admin     root      true     2        1
-newgroup  testuser  false    118      110
+role      member    isAdmin
+admin     newgroup  false
+admin     root      true
+newgroup  testuser  false
 
 
 user root
@@ -740,12 +740,12 @@ CREATE TABLE db2.s1.foo (k int);
 
 user root
 
-query TTBOO colnames,rowsort
-SELECT * FROM system.role_members
+query TTB colnames,rowsort
+SELECT role, member, "isAdmin" FROM system.role_members
 ----
-role      member    isAdmin  role_id  member_id
-admin     root      true     2        1
-newgroup  testuser  false    118      110
+role      member    isAdmin
+admin     root      true
+newgroup  testuser  false
 
 statement ok
 GRANT ALL ON DATABASE db2 TO newgroup
@@ -825,11 +825,11 @@ DROP TABLE db2.foo
 statement ok
 DROP USER testuser
 
-query TTBOO colnames
-SELECT * FROM system.role_members
+query TTB colnames
+SELECT role, member, "isAdmin" FROM system.role_members
 ----
-role   member  isAdmin  role_id  member_id
-admin  root    true     2        1
+role   member  isAdmin
+admin  root    true
 
 statement error cannot drop role/user newgroup: grants still exist on db2
 DROP ROLE newgroup
@@ -883,11 +883,11 @@ REVOKE admin FROM public
 statement ok
 CREATE USER testuser
 
-query TTBOO colnames
-SELECT * FROM system.role_members
+query TTB colnames
+SELECT role, member, "isAdmin" FROM system.role_members
 ----
-role   member  isAdmin  role_id  member_id
-admin  root    true     2        1
+role   member  isAdmin
+admin  root    true
 
 
 user testuser
@@ -962,11 +962,11 @@ CREATE ROLE IF NOT EXISTS anotherrolewithcreate2 CREATEROLE
 statement ok
 CREATE ROLE IF NOT EXISTS rolewithoutcreate2 WITH NOCREATEROLE
 
-query TTBOO colnames
-SELECT * FROM system.role_members
+query TTB colnames
+SELECT role, member, "isAdmin" FROM system.role_members
 ----
-role   member  isAdmin  role_id  member_id
-admin  root    true     2        1
+role   member  isAdmin
+admin  root    true
 
 user testuser
 

